images/archive/reference.go

```diff
@@ -69,7 +69,7 @@ func isImagePrefix(s, prefix string) bool {
 
 func normalizeReference(ref string) (string, error) {
 	// TODO: Replace this function to not depend on reference package
```

Comment:
I think this comment makes us not want to depend on this package?

Reply:
Confusing given that the CRI utility function definitely depends on the reference package as well. @dmcgowan ?

Reply:
Yes, this is not the "final state" of this I guess, but this was at least a step to prevent introducing an additional dependency in the CLI.

From https://github.com/docker/distribution/pull/2786#discussion_r240436599

> Just as a note, I want to split the reference package out of this repository and explicitly have a `Docker` referencing package for all projects to use which make use of this referencing logic. Ironically this reference package is least useful to this repository since the reference format used by the registry can be vastly simplified as a superset of what Docker uses. I don't have time to refactor this, but it is not harmful to have this functionality along side all the other Docker-specific normalization functions.

Reply:
I still agree with this comment, but I also agree with @thaJeztah about cutting out the middleman dependency. This makes it much more clear where we need to replace this functionality, in both CRI and containerd.